tests/testsuite/env.rs

```diff
@@ -518,6 +518,117 @@ fn test_parse_string_and_list_ignore_list_parse_key_case() {
     );
 }
 
+#[test]
+fn test_parse_nested_double_separator() {
```

Comment:
without kebab case conversion, this is to ensure it works in simple mode as a mirror of the next test which enables single underscore nesting